src/platform/sdl2-platform.c

```diff
@@ -487,12 +491,29 @@ static void _notifyEvent(short eventId, int data1, int data2, const char *str1,
     //Unused
 }
 
+/*
+ * Take screenshot in current working directory (ScreenshotN.png)
+ */
+static boolean _takeScreenshot() {
+    char screenshotFilepath[BROGUE_FILENAME_MAX];
+
+    getAvailableFilePath(screenshotFilepath, "Screenshot", SCREENSHOT_SUFFIX);
+    strcat(screenshotFilepath, SCREENSHOT_SUFFIX);
+
+    if (WinSurf) {
+        IMG_SavePNG(WinSurf, screenshotFilepath);
+        return true;
+    }
+    return false;
+}
+
 struct brogueConsole sdlConsole = {
     _gameLoop,
     _pauseForMilliseconds,
     _nextKeyOrMouseEvent,
     _plotChar,
     _remap,
     _modifierHeld,
-    _notifyEvent
+    _notifyEvent,
+    _takeScreenshot
```

Comment:
You'll need to add this to curses and web consoles, too. What I recommend is you change the definition of takeScreenshot to be something like this:

```c
boolean takeScreenshot() {
    if (currentConsole.takeScreeshot) {
        return currentConsole.takeScreeshot();
    } else {
        return false;
    }
}
```

Then I think you can set the function pointer to NULL in the other consoles.

Reply:
I see you've added dummy implementations to the curses and web consoles - that works too, but it means this code change ^ is unnecessary. Could you either have this if statement and NULL in the consoles, or no if statement and the dummy functions?